### PR TITLE
Not all class members are proposed in completion

### LINKED ISSUE #194: Not all class members are proposed in completion
If statement with class members (e.g. method or field) is finished then invoking completion inside such element will not return all possible members with prefix before cursor. Example:
```php
$a = new TestClass();
$a->test|Method();
```
[TestClass](https://github.com/felixfbecker/php-language-server/blob/master/fixtures/global_symbols.php#L21) is from fixtures. If completion will be triggered in place of `|` then only `testMethod()` is proposed. Property `testProperty` should be also visible.

### PR DESCRIPTION
**Pull request by mniewrzal**
Closes #194

### CODE REVIEW

File: src/CompletionProvider.php [hunk 1]
```diff
@@ -156,8 +154,12 @@ public function provideCompletion(PhpDocument $doc, Position $pos): CompletionLi
                     }
                 }
             } else {
-                $fqn = $this->definitionResolver->resolveReferenceNodeToFqn($node);
-                $prefixes = $fqn !== null ? [$fqn] : [];
+                $prefix = $this->definitionResolver->resolveReferenceNodeToFqn($node);
+                $index = ($index = strpos($prefix, '->')) ? $index : strpos($prefix, '::');
```
Comment (felixfbecker) [inline src/CompletionProvider.php]: `$prefix` can be `null`, in which case the call to `strpos` is incorrect. This is why I saved it in the variable `$fqn`.
Also I find this very hard to read, I would just do `preg_split('/->|::/', $fqn)[0]`.
But tbh I think if you don't want to handle the case of already having typed a prefix, the check at line 137|135 can just go away and this `else` branch is not needed - it exists only to filter by an already-typed property name.

Comment (mniewrzal) [inline src/CompletionProvider.php]: Good point. Thanks for hint!

File: src/CompletionProvider.php [hunk 2]
```diff
@@ -134,30 +132,25 @@ public function provideCompletion(PhpDocument $doc, Position $pos): CompletionLi
...
-            if (!is_string($node->name)) {
-                // If the name is an Error node, just filter by the class
+            // If the name is an Error node, just filter by the class
+            if ($node instanceof Node\Expr\MethodCall || $node instanceof Node\Expr\PropertyFetch) {
+                // For instances, resolve the variable type
+                $prefixes = DefinitionResolver::getFqnsFromType(
+                    $this->definitionResolver->resolveExpressionNodeToType($node->var)
+                );
+            } else {
+                $prefixes = [$node->class instanceof \PhpParser\Node\Name ? (string)$node->class : ''];
```
Comment (felixfbecker) [inline src/CompletionProvider.php]: Why this change? `Node` is used as a shorthand everywhere in the codebase

Comment (mniewrzal) [inline src/CompletionProvider.php]: For some reason previous condition returns false for case like 'TestClass::$st'. Most probably because there is namespace \PhpParser\Node\Name and there is class 'Name' in namespace `\PhpParser\Node`. I change it and I wanted to ask what will be the best solution for this :) You were faster.

Comment (felixfbecker) [inline src/CompletionProvider.php]: PhpParser does this pattern a lot - defining an abstract base class like `PhpParser\Node\Expr` and then several implementations in a namespace like `PhpParser\Node\Expr\Variable`. This works fine in all other cases. Could you check the actual value of `$node->class` in your error case?

Comment (mniewrzal) [inline src/CompletionProvider.php]: `$node->class` is `PhpParser\Node\Name\FullyQualified`. `(string)$node->class` gives proper class name.

Comment (felixfbecker) [inline src/CompletionProvider.php]: All tests pass when using `Node\Name`

Comment (mniewrzal) [inline src/CompletionProvider.php]: Hmm strange... but good that its not a problem in general. Thanks for help.
